### PR TITLE
refactor: replace char_u with char

### PR DESCRIPTION
**Pull request by dundargoc**
Work on https://github.com/neovim/neovim/issues/459

### CODE REVIEW

File: src/nvim/fileio.h [hunk 1]
```diff
@@ -18,7 +18,7 @@
 #define READ_NOWINENTER 0x80    // do not trigger BufWinEnter
 #define READ_NOFILE     0x100   // do not read a file, do trigger BufReadCmd
 
-#define READ_STRING(x, y) (char_u *)read_string((x), (size_t)(y))
+#define READ_STRING(x, y) read_string((x), (size_t)(y))
```
Comment (zeertzjq) [inline src/nvim/fileio.h:21]: This macro can be removed. It's not present in Vim either.

Comment (dundargoc) [reply src/nvim/fileio.h:21]: I get an error when I try to remove it. It seems to be used in spellfile.c.

Comment (dundargoc) [reply src/nvim/fileio.h:21]: Oh you mean just substitute the macro, gotcha.

File: src/nvim/textformat.c [hunk 2]
```diff
@@ -458,7 +458,7 @@ void internal_format(int textwidth, int second_indent, int flags, bool format_on
   }
 
   if (save_char != NUL) {               // put back space after cursor
-    pchar_cursor((char_u)save_char);
+    pchar_cursor((char)(uint8_t)save_char);
```
Comment (zeertzjq) [inline src/nvim/textformat.c:461]: Better to change `save_char` to `char`

File: src/nvim/textformat.c [hunk 3]
```diff
@@ -93,7 +93,7 @@ void internal_format(int textwidth, int second_indent, int flags, bool format_on
       && !(State & VREPLACE_FLAG)) {
     cc = gchar_cursor();
     if (ascii_iswhite(cc)) {
-      save_char = cc;
+      save_char = (char)(uint8_t)cc;
```
Comment (zeertzjq) [inline src/nvim/textformat.c:96]:
```suggestion
      save_char = (char)cc;
```

File: src/nvim/option.c [hunk 4]
```diff
@@ -1963,7 +1963,7 @@ static char *set_bool_option(const int opt_idx, char_u *const varp, const int va
     // delete the undo file, the option may be set again without making
     // any changes in between.
     if (curbuf->b_p_udf || p_udf) {
-      char_u hash[UNDO_HASH_SIZE];
+      char hash[UNDO_HASH_SIZE];
```
Comment (zeertzjq) [inline src/nvim/option.c:1966]:
```suggestion
      uint8_t hash[UNDO_HASH_SIZE];
```

File: src/nvim/option.c [hunk 5]
```diff
@@ -1973,8 +1973,8 @@ static char *set_bool_option(const int opt_idx, char_u *const varp, const int va
         if ((curbuf == bp
              || (opt_flags & OPT_GLOBAL) || opt_flags == 0)
             && !bufIsChanged(bp) && bp->b_ml.ml_mfp != NULL) {
-          u_compute_hash(bp, hash);
-          u_read_undo(NULL, hash, bp->b_fname);
+          u_compute_hash(bp, (char_u *)hash);
```
Comment (zeertzjq) [inline src/nvim/option.c:1976]:
```suggestion
          u_compute_hash(bp, hash);
```

File: src/nvim/option.c [hunk 6]
```diff
@@ -1973,8 +1973,8 @@ static char *set_bool_option(const int opt_idx, char_u *const varp, const int va
...
+          u_read_undo(NULL, (char_u *)hash, bp->b_fname);
```
Comment (zeertzjq) [inline src/nvim/option.c:1977]:
```suggestion
          u_read_undo(NULL, hash, bp->b_fname);
```

File: src/nvim/search.c [hunk 7]
```diff
@@ -3104,7 +3104,7 @@ static int fuzzy_match_recursive(const char *fuzpat, const char *str, uint32_t s
 /// normalized and varies with pattern.
 /// Recursion is limited internally (default=10) to prevent degenerate cases
 /// (pat_arg="aaaaaa" str="aaaaaaaaaaaaaaaaaaaaaaaaaaaaaa").
-/// Uses char_u for match indices. Therefore patterns are limited to
+/// Uses char for match indices. Therefore patterns are limited to
 /// MAX_FUZZY_MATCHES characters.
```
Comment (zeertzjq) [inline src/nvim/search.c:3108]:
```suggestion
/// Patterns are limited to MAX_FUZZY_MATCHES characters.
```

File: src/nvim/spell.c [hunk 8]
```diff
@@ -2652,24 +2652,24 @@ void onecap_copy(char *word, char *wcopy, bool upper)
...
     int c = mb_cptr2char_adv((const char **)&s);
 
     if (c == 0xdf) {
       c = 'S';
-      if (d - (char_u *)wcopy >= MAXWLEN - 1) {
+      if (d - wcopy >= MAXWLEN - 1) {
         break;
       }
-      *d++ = (char_u)c;
+      *d++ = (char)(uint8_t)c;
```
Comment (zeertzjq) [inline src/nvim/spell.c:2664]:
```suggestion
      *d++ = (char)c;
```

**Pull request merged**